qualang_tools/addons/calibration/README.md

```diff
@@ -0,0 +1,167 @@
+options = {
+    "fontsize": 14,
+    "color": "b",
+    "marker": ".",
+    "linewidth": 1,
+    "figsize": (12, 15),
+}
+#############################################################
+
+# Initialize the calibration API with the relevant configuration parameters
+my_calib = QUA_calibrations(
+    configuration=config,
+    readout=(resonator_element, resonator_operation),
+    qubit=(qubit_element, qubit_operation),
+    integration_weights=int_w,
+    outputs=outs,
+)
+
+# Set calibrations from the ones detailed below
+my_calib.set_...
+
+# Open quantum machines manager
+qmm = QuantumMachinesManager(host="127.0.0.1", port=80)
+
+# Simulate the calibrations
+my_calib.simulate_calibrations(qmm, simulation_duration=5000)
+
+# Run the calibrations
+qm = qmm.open_qm(config)
+my_calib.run_calibrations(quantum_machine=qm, plot="full", plot_options=options)
+```
+
+## Available calibrations
+### Raw traces
+This program acquires the raw and averaged traces from the two inputs of the OPX.
+It only requires two inputs:
+* `iterations`: number of averaging iterations. Must be a positive Python integer.
+* `cooldown_time`: wait time between two iterations in clock cycle unit (4ns). Must be an integer.
+```python
+my_calib.set_raw_traces(iterations=1000, cooldown_time=0)
+```
+![raw_traces](raw_traces.PNG)
+
+### Time of flight
+This program measures the time of flight and offsets of the two inputs of the OPX.
+It requires three inputs:
+* `iterations`: number of averaging iterations. Must be a positive Python integer.
+* `cooldown_time`: wait time between two iterations in clock cycle unit (4ns). Must be an integer.
+* `threshold`: detection threshold for time of flight estimation. The default value is 10/4096 which is usually good enough to detect the waveform.
+```python
+my_calib.set_time_of_flight(iterations=1000, cooldown_time=0, threshold=10/4096)
+```
+![time_of_flight](time_of_flight.PNG)
+
+### Resonator spectroscopy
+This program measures the resonator response for different readout pulse frequencies defined in scan_variables.
+It requires three inputs:
+* `scan_variables`: variables to be scanned. Must be an array of tuples [('variable', [values])]. 1D and 2D scans are available.
+* `iterations`: number of averaging iterations. Must be a positive Python integer.
+* `cooldown_time`: resonator cooldown time in clock cycles (4ns). Must be an integer.
+
+```python
+scans = [("frequency", np.arange(100e6, 300e6, step=0.1e6))]
+my_calib.set_resonator_spectroscopy(scan_variables=scans, iterations=1000, cooldown_time=10_000//4)
+```
+
+### Rabi measurement
+This program performs a Rabi-like measurement by sending a pulse to the qubit and measuring the resonator response.
+
+It requires three inputs:
+* `scan_variables`: variables to be scanned. Must be an array of tuples [('variable', [values])]. 1D and 2D scans are available and the scanning parameters are:
+  * 'duration': pulse length in clock cycles (4ns), must be an integer.
```

Comment:
If the pulse is arbitrary then the duration must be longer than it. Maybe we can also assert that?

Reply:
Very good point to avoid what happened at IST...